src/main/java/com/google/firebase/internal/FirebaseExecutors.java

```diff
@@ -55,11 +55,10 @@
 
     private final Object lock = new Object();
     private final Set<String> apps = new HashSet<>();
-
-    private ScheduledExecutorService executorService;
+    private ExecutorService executorService;
 
     @Override
-    protected ScheduledExecutorService getExecutor(FirebaseApp app) {
+    protected synchronized ExecutorService getExecutor(FirebaseApp app) {
```

Comment:
Don't lock on the main object (and don't lock twice).

Reply:
Ah! Missed it during the merge with #74. Thanks for pointing that out.